client/web/src/enterprise/codeintel/detail/CodeIntelIndexTimeline.tsx

```diff
@@ -0,0 +1,265 @@
+                    <tr>
+                        <th className="docker-command-spec__header">Image</th>
+                        <td>{meta.image}</td>
+                    </tr>
+                    <tr>
+                        <th className="docker-command-spec__header">Commands</th>
+                        <td>
+                            <code>{meta.commands.join(' ')}</code>
+                        </td>
+                    </tr>
+                    <tr>
+                        <th className="docker-command-spec__header">Root</th>
+                        <td>/{meta.root}</td>
+                    </tr>
+                </table>
+            )}
+        </div>
+
+        <div className="p-2">
+            {logEntry.out ? (
+                <Collapsible title="Log output" titleAtStart={true} buttonClassName="p-2">
+                    <LogOutput text={logEntry.out} />
+                </Collapsible>
+            ) : (
+                <div className="p-2">
+                    <span className="text-muted">No log output available.</span>
+                </div>
+            )}
+        </div>
+    </div>
+)
+
+interface LogOutputProps {
+    text: string
+    className?: string
+}
+
+const LogOutput: FunctionComponent<LogOutputProps> = ({ text, className }) => (
+    <pre className={classNames('code-intel-index-timeline-log-output', 'bg-code rounded p-3 mb-0', className)}>
+        {
+            // Use index as key because log lines may not be unique. This is OK
+            // here because this list will not be updated during this component's
+            // lifetime.
+            /* eslint-disable react/no-array-index-key */
+            text.split('\n').map((line, index) => (
+                <code
+                    key={index}
+                    className={classNames(
+                        'code-intel-index-timeline-log-output__line',
+                        line.startsWith('stdout:') ? 'stdout' : line.startsWith('stderr:') ? 'stderr' : ''
+                    )}
+                >
+                    {line.replace(/^std(out|err): /, '')}
+                </code>
+            ))
+        }
+    </pre>
+)
+
+const timeOrders: [number, string][] = [
+    [1000 * 60 * 60 * 24, 'day'],
+    [1000 * 60 * 60, 'hour'],
+    [1000 * 60, 'minute'],
+    [1000, 'second'],
+    [1, 'millisecond'],
+]
+
+const formatMilliseconds = (milliseconds: number): string => {
+    const parts: string[] = []
+
+    // Construct a list of parts like `1 day` or `7 hours` in descending
+    // order. If the value is zero, an empty string is added to the list.`
+    timeOrders.reduce((msRemaining, [denominator, suffix]) => {
+        // Determine how many units can fit into the current value
+        const part = Math.floor(msRemaining / denominator)
+        // Format this part (pluralize if value is more than one)
+        parts.push(part > 0 ? `${part} ${suffix}${part > 1 ? 's' : ''}` : '')
```

Comment:
```suggestion
        parts.push(part > 0 ? `${part} ${pluralize('part', part)}` : '')
```

seems a bit easier to read to me, but feel free to disregard